driver/src/orderbook/streamed/balance.rs

```diff
@@ -2,17 +2,43 @@ use super::*;
 use anyhow::{anyhow, ensure, Result};
 use serde::{Deserialize, Serialize};
 
+/// Balance change from a potential solution that might still get replaced by a better solution in
+/// the same batch.
+#[derive(Clone, Copy, Debug, Default, Deserialize, Serialize)]
+struct Proceeds {
+    batch_id: BatchId,
+    // Two fields instead of one I256 because although unlikely it could be possible to overflow a
+    // I256 with multiple trades in one batch.
+    increase: U256,
+    decrease: U256,
+}
+
+enum TradeType {
+    Sell,
+    Buy,
+}
+
+impl Proceeds {
+    fn get_field(&mut self, operation: TradeType) -> &mut U256 {
+        match operation {
+            TradeType::Sell => &mut self.decrease,
+            TradeType::Buy => &mut self.increase,
+        }
+    }
+}
+
 /// The balance of a token for a user.
 #[derive(Clone, Copy, Debug, Default, Deserialize, Serialize)]
 pub struct Balance {
     balance: U256,
     deposit: Flux,
     withdraw: Flux,
+    proceeds: Proceeds,
```

Comment:
Just curious, what is the reason these can't be applied to the balance immediately? Is it because balances can temporarily drop below 0 during settlement?

Reply:
Yes. I think Felix asked the same question previously and I forgot to move the comment here. You can in the same batch buy a token and sell it for a balance that includes the new buy amount and is larger than the balance without the new buy. That is how I understood the smart contract code.

Reply:
Makes sense.